web/user_accounts/admin.py

```
@@ -54,3 +59,71 @@ class EthereumAccountAdmin(admin.ModelAdmin):
 
 
 admin.site.register(EthereumAccount, EthereumAccountAdmin)
+
+
+def si_approve(self, request, queryset):
+    affected_rows = queryset.update(approved=True)
+    for object in queryset:
+        LogEntry.objects.log_action(
+            user_id=request.user.pk,
+            content_type_id=ContentType.objects.get_for_model(object).pk,
+            object_id=object.pk,
+            object_repr=str(object),
+            action_flag=CHANGE,
+            change_message=_('Approved')
+        )
+    self.message_user(
+        request,
+        "%s sophisticated investor document(s) approved" % affected_rows)
+    email_content = render_to_string(
+        'user_accounts/email/si_approved.txt'
+    )
+    send_mail(
+        subject='Approved - Sophisticated Investor check',
+        message=email_content,
+        from_email=settings.DEFAULT_FROM_EMAIL,
+        recipient_list=[request.user.email],
+        fail_silently=True,
+    )
+
+
+si_approve.short_description = "Approve selected sophisticated investors"
+
+
+def si_decline(self, request, queryset):
+    affected_rows = queryset.update(approved=False)
+    for object in queryset:
+        LogEntry.objects.log_action(
+            user_id=request.user.pk,
+            content_type_id=ContentType.objects.get_for_model(object).pk,
+            object_id=object.pk,
+            object_repr=str(object),
+            action_flag=CHANGE,
+            change_message=_('Declined')
+        )
+    self.message_user(
+        request,
+        "%s sophisticated investor document(s) declined" % affected_rows)
+    email_content = render_to_string(
+        'user_accounts/email/si_declined.txt'
+    )
+    send_mail(
```

Can we not use fail silently here? I'd rather report something to sentry than have it gobbled up.

---

Fixed up to log an error ... which in turn gets reported to sentry.